src/types/datetime.rs

```diff
@@ -486,6 +486,39 @@ pub fn timezone_utc(py: Python<'_>) -> &PyTzInfo {
     unsafe { &*(ensure_datetime_api(py).TimeZone_UTC as *const PyTzInfo) }
 }
 
+/// Equivalent to `datetime.timezone`
+// Not making this public yet as the implementation is slightly different from
+// original cpython.
+// #[crate::pyclass]
+// pub(crate) struct PyTimeZone {
+//     offset: PyDelta,
+//     // name
+// }
```

Comment:
I was thinking it is better to have our own timezone port, but I haven't figured out how to get `pyclass` working yet. Original source https://github.com/python/cpython/blob/29f1b0bb1ff73dcc28f0ca7e11794141b6de58c9/Lib/datetime.py#L2296-L2314

Reply:
I had the same problem here, it's the first time I work in PyO3, but it seems those macros are never used inside the crate, I think this will have to be done without the macros

Reply:
The problem with having a pyclass such as this is that they behave poorly when shared between libraries. See also https://github.com/PyO3/pyo3/issues/1444

Most likely you (or your users) will run into weird errors like `TypeError: 'PyTimeZone' object cannot be converted to 'PyTimeZone'`.

Reply:
Just read that issue, it makes sense, so we won't have this here and we should ask users to implement the type in their own crate. We could add the example to the docs pointing at that issue, so that if it's fixed in the future we can move the PyClass here.